.drone.yml

```diff
@@ -0,0 +1,199 @@
+  depends_on:
+  - mod-download
+
+- name: build
+  image: golang:1.12.7
+  commands:
+  - go build -o gcs-emulator -mod readonly ./cmd/gcs-emulator
+  depends_on:
+  - mod-download
+
+- name: test-ci-dockerfile
+  image: plugins/docker
+  settings:
+    dockerfile: drone/Dockerfile
+    dry_run: true
+    repo: fsouza/gcs-emulator
+  when:
+    event:
+    - pull_request
+  depends_on:
+  - build
+
+- name: test-dockerfile
+  image: plugins/docker
+  settings:
+    dry_run: true
+    repo: fsouza/gcs-emulator
+  when:
+    event:
+    - pull_request
+  depends_on:
+  - clone
+
+- name: build-and-push-to-dockerhub
+  image: plugins/docker
+  settings:
+    auto_tag: true
+    dockerfile: drone/Dockerfile
+    password:
+      from_secret: docker_password
+    repo: fsouza/gcs-emulator
+    username:
+      from_secret: docker_username
+  when:
+    ref:
+    - "refs/tags/*"
+    - refs/heads/master
+  depends_on:
+  - test
+  - lint
+  - build
+
+---
+kind: pipeline
+name: build_go_1.11.12
+
+platform:
+  os: linux
+  arch: amd64
+
+workspace:
+  base: /go
+  path: gcs-emulator
+
+steps:
+- name: mod-download
+  image: golang:1.11.12
+  commands:
+  - go mod download
+  environment:
+    GOPROXY: https://proxy.golang.org
+  depends_on:
+  - clone
+
+- name: test
+  image: golang:1.11.12
+  commands:
+  - go test -race -vet all -mod readonly ./...
+  depends_on:
+  - mod-download
+
+- name: lint
+  pull: always
+  image: golangci/golangci-lint
+  commands:
+  - golangci-lint run --enable-all -D errcheck -D lll -D dupl -D gochecknoglobals -D unparam --deadline 5m ./...
+  depends_on:
+  - mod-download
+
+- name: build
+  image: golang:1.11.12
+  commands:
+  - go build -o gcs-emulator -mod readonly ./cmd/gcs-emulator
+  depends_on:
+  - mod-download
+
+- name: test-ci-dockerfile
+  image: plugins/docker
+  settings:
+    dockerfile: drone/Dockerfile
+    dry_run: true
+    repo: fsouza/gcs-emulator
+  when:
+    event:
+    - pull_request
+  depends_on:
+  - build
+
+---
+kind: pipeline
+name: build_go_1.13beta1
+
+platform:
+  os: linux
+  arch: amd64
+
+workspace:
+  base: /go
+  path: gcs-emulator
+
+steps:
+- name: mod-download
+  image: golang:1.13beta1
+  commands:
+  - go mod download
+  environment:
+    GOPROXY: https://proxy.golang.org
+  depends_on:
+  - clone
+
+- name: test
+  image: golang:1.13beta1
+  commands:
+  - go test -race -vet all -mod readonly ./...
+  depends_on:
+  - mod-download
+
+- name: lint
+  pull: always
+  image: golangci/golangci-lint
+  commands:
+  - golangci-lint run --enable-all -D errcheck -D lll -D dupl -D gochecknoglobals -D unparam --deadline 5m ./...
+  depends_on:
+  - mod-download
+
+- name: build
+  image: golang:1.13beta1
+  commands:
+  - go build -o gcs-emulator -mod readonly ./cmd/gcs-emulator
+  depends_on:
+  - mod-download
+
+- name: test-ci-dockerfile
+  image: plugins/docker
+  settings:
+    dockerfile: drone/Dockerfile
```

Comment:
I'm not sure how Drone works, but I tried this:

```
# mod-download?
go mod download
# build?
go build -o gcs-emulator -mod readonly ./cmd/gcs-emulator
# test-ci-dockerfile?
docker build . --file drone/Dockerfile --tag plugins/docker:test
# Run it?
docker run plugins/docker:test
```
This results in:
```
standard_init_linux.go:207: exec user process caused "exec format error"
```

The docker image created from the root ``Dockerfile`` does appear to do the right thing.

Reply:
@jwhitlock I assume you're on a Mac? `go build -o gcs-emulator -mod readonly ./cmd/gcs-emulator` generates a Mach-O binary runnable on Mac and Docker runs on Linux, so the command would need to be something like ``GOOS=linux go build -o gcs-emulator -mod readonly ./cmd/gcs-emulator`` to get go to cross compile it (drone doesn't need to set GOOS=linux because it's also running on Linux, see https://cloud.drone.io/fsouza/fake-gcs-server/20/1/6).

But let's move the conversation to #47, as that seems to be a more complete solution! I can move it from travis to drone independently or push the changes there.

Reply:
Good guess! Switching to ``GOOS=linux go build -o gcs-emulator -mod readonly ./cmd/gcs-emulator`` make this work for me.